### DISCUSSION TITLE
Invalid command name "tee"

### DISCUSSION THREAD

**Discussion by shivaraja24** [siliconcompiler/siliconcompiler#3080] [Development]
Using the latest 0.29.0 version I am getting the following error message when at the end of floorplan.init phase.
| INFO    | job0 | floorplan.init       | 0 | tee -file reports/dont_touch.start.rpt {report_dont_touch}
| INFO    | job0 | floorplan.init       | 0 | Error: preamble.tcl, 94 invalid command name "tee"
This is probably the cause to fail generation of <top>.sdc due to which the next phase 'floorplan.macro_placement' errors out.

Going through the pull-requests, it seems the recent change, #3042, updated the file siliconcompiler/tools/openroad/scripts/apr/preamble.tcl, to include the following lines,

tee -file reports/dont_touch.start.rpt {report_dont_touch}
tee -file reports/dont_use.start.rpt {report_dont_use}

Any reason why ? Is there an overriding 'tee' command elsewhere as the tcl/shell prompt based 'tee' does not support -file option.

**Comment by gadfort** [comment]
Do you have `novercheck` turned on? The minimum version of OpenROAD should support that command.
What OpenROAD version are you using?

**Comment by shivaraja24** [comment]
Am using OpenRoad version v2.0-15587-g834b7a0fb. And i did check that standalone OpenRoad also gives this error.

openroad> tee -file x {xyz}
invalid command name "tee"
while evaluating tee -file x {xyz}

**Reply by gadfort** [reply]
siliconcompiler should have complained that this OpenROAD version is too old. Have you turned off version checking via `option,novercheck`?

You will need to update OpenROAD since that is a newer command in OpenROAD.

**Reply by shivaraja24** [reply]
Thanks. SC did complain and i overwrote it with the following. Which version would you recommend to use ?

    chip.set('tool', 'openroad', 'version', '>=v2.0-15587')

**Reply by gadfort** [reply]
the scripts are written for `>=v2.0-17598`, so I would suggest that or newer. If you want to override the version, then you will need to modify the scripts as well.

You can use `sc-install openroad` to install the currently tested version of openroad.

**Reply by shivaraja24** [reply]
Any version later than OpenRoad v2.0-17598 may be ok? I will try with the later versions and update. Meanwhile, please do close this issue.
Thanks.

**Reply by gadfort** [reply]
Sounds good. let me know if you encounter any new issues.

**Comment by shivaraja24** [comment]
Though i see openroad in the list, from the source code sc_install.py, running 'sc-install openroad', gives the following error. What am I missing ? Also the OpenRoad 24Q3 tag installed from (https://github.com/The-OpenROAD-Project/OpenROAD/tags) included v2.0-15587.

$ sc-install openroad
usage: sc-install [-h]
                  [-group {digital-simulation,analog-simulation} [{digital-simulation,analog-simulation} ...]]
                  [-prefix <path>] [-build_dir <path>] [-show] [-debug_machine]
                  [{magic,netgen,xyce,slang,sv2v,klayout,montage,ghdl,yosys,verible,chisel,verilator,icarus,surelog} ...]
sc-install: error: argument tool: invalid choice: 'openroad' (choose from 'magic', 'netgen', 'xyce', 'slang', 'sv2v', 'klayout', 'montage', 'ghdl', 'yosys', 'verible', 'chisel', 'verilator', 'icarus', 'surelog')

**Reply by gadfort** [reply]
it means you are on a system which we don't support building openroad for unfortunately. Usually RHEL8.

**Reply by shivaraja24** [reply]
Thanks

**Reply by shivaraja24** [reply]
Hi Peter, We use AlmaLinux 8.10 which is equivalent to RHEL8.10. Are there plans to support openroad for RHEL8 ?

**Reply by gadfort** [reply]
They plan to support rhel8: https://github.com/The-OpenROAD-Project/OpenROAD/issues/5951 (it used to work, but the current dependencies are not available on rhel8 as far as I can tell)
I would suggest pinging them about to, to indicate interest in this support.

**Reply by shivaraja24** [reply]
Thanks.
